sycl/include/CL/sycl/detail/defines_elementary.hpp

```diff
@@ -48,3 +48,24 @@
 #else
 #define __SYCL_INLINE_CONSTEXPR static constexpr
 #endif
+
+#ifndef __SYCL_HAS_CPP_ATTRIBUTE
+#if defined(__cplusplus) && defined(__has_cpp_attribute)
+#define __SYCL_HAS_CPP_ATTRIBUTE(x) __has_cpp_attribute(x)
+#else
+#define __SYCL_HAS_CPP_ATTRIBUTE(x) 0
+#endif
+#endif
+
+#if defined(__cplusplus) && __cplusplus > 201402L &&                           \
```

Comment:
late nit:
```suggestion
#ifndef __SYCL_FALLTHROUGH
#if defined(__cplusplus) && __cplusplus > 201402L &&                           \
```

Reply:
Fixed in https://github.com/intel/llvm/pull/2980/commits/a60e63f1c597cd83e588fe0cf5337929998ba83f